File: backend/src/main/java/tom/projects/monkeylog/repository/workout/WorkoutRepository.java
package tom.projects.monkeylog.repository.workout;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import tom.projects.monkeylog.model.workout.Workout;
import tom.projects.monkeylog.model.workout.WorkoutType;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public interface WorkoutRepository extends JpaRepository<Workout, UUID> {
    List<Workout> findAllByWorkoutTypeAndUserId(WorkoutType workoutType, UUID userId);

    Page<Workout> findAllByWorkoutTypeAndUserId(WorkoutType workoutType, UUID userId, Pageable pageable);

    Page<Workout> findAllByWorkoutTypeAndUserIdAndStartDateAfter(WorkoutType workoutType, UUID userId, LocalDateTime start, Pageable pageable);
}

File: backendkt/src/main/kotlin/tom/com/monkeylog/repository/workout/WorkoutRepository.kt
package tom.com.monkeylog.repository.workout

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import tom.com.monkeylog.model.workout.Workout
import tom.com.monkeylog.model.workout.WorkoutType
import java.time.LocalDateTime
import java.util.*

interface WorkoutRepository : JpaRepository<Workout, UUID> {
    fun findAllByWorkoutTypeAndUserId(workoutType: WorkoutType, userId: UUID): List<Workout>
    fun findAllByWorkoutTypeAndUserId(workoutType: WorkoutType, userId: UUID, pageable: Pageable): Page<Workout>
    fun findAllByWorkoutTypeAndUserIdAndStartDateAfter(
        workoutType: WorkoutType,
        userId: UUID,
        start: LocalDateTime,
        pageable: Pageable
    ): Page<Workout>
}
